play-services-auth-api-phone/core/src/main/kotlin/org/microg/gms/auth/phone/SmsRetrieverCore.kt

```diff
@@ -246,12 +246,16 @@ class SmsRetrieverCore(private val context: Context, override val lifecycle: Lif
         override fun onReceive(context: Context, intent: Intent) {
             if (Telephony.Sms.Intents.SMS_RECEIVED_ACTION == intent.action) {
                 val messages = Telephony.Sms.Intents.getMessagesFromIntent(intent)
+                val messageBodyBuilder = StringBuilder()
+                var senderPhoneNumber: String? = null
                 for (message in messages) {
-                    try {
-                        handleIncomingSmsMessage(message)
-                    } catch (e: Exception) {
-                        Log.w(TAG, "Error handling incoming SMS", e)
-                    }
+                    messageBodyBuilder.append(message.messageBody)
+                    senderPhoneNumber = message.originatingAddress
```

Comment:
@mar-v-in
I don't this code but wouldn't be safer to check that all messages come from the same `originatingAddress`?

Reply:
The `SMS_RECEIVED_ACTION` broadcast is sent and handled for a single logical SMS. Long logical SMS for technical reasons are split into multiple PDUs, that result in multiple `SmsMessage` instances in Android. As such there can't be multiple senders of the message objects here.